lib/geometry/polygon.nit

```diff
@@ -0,0 +1,428 @@
+			var p2 = new Point[Float](points[(i+1) % points.length].x, points[(i+1) % points.length].y)
+			var edge = new Point[Float](p1.x - p2.x, p1.y - p2.y)
+			var normal = new Point[Float](-edge.y, edge.x)
+			axes[i] = normal
+		end
+		return axes
+	end
+
+	# Sort the vertices in counter clockwise order
+	# ~~~
+	# var p1 = new Point[Float](0.0, 0.0)
+	# var p2 = new Point[Float](5.0, 0.0)
+	# var p3 = new Point[Float](0.0, 5.0)
+	# var p4 = new Point[Float](5.0, 5.0)
+	# var arr = new Array[Point[Float]].with_items(p1, p2, p3, p4)
+	# var poly = new ConvexPolygon.with_vertices(arr)
+	# poly.anti_clock_sort
+	# assert poly.points == [p4, p2, p1, p3]
+	# ~~~
+	fun anti_clock_sort do
+		var sorter = new AntiClockSort.with_center(vertices)
+		sorter.sort(points)
+	end
+
+	# Sort the vertices in clockwise order
+	# ~~~
+	# var p1 = new Point[Float](0.0, 0.0)
+	# var p2 = new Point[Float](5.0, 0.0)
+	# var p3 = new Point[Float](0.0, 5.0)
+	# var p4 = new Point[Float](5.0, 5.0)
+	# var arr = new Array[Point[Float]].with_items(p1, p2, p3, p4)
+	# var poly = new ConvexPolygon.with_vertices(arr)
+	# poly.clock_sort
+	# assert poly.points == [p3, p1, p2, p4]
+	# ~~~
+	fun clock_sort do
+		var sorter = new ClockSort.with_center(vertices)
+		sorter.sort(points)
+	end
+
+	# Does this polygon intersects `other` ?
+	# ~~~
+	# var p1 = new Point[Float](0.0, 0.0)
+	# var p2 = new Point[Float](5.0, 0.0)
+	# var p3 = new Point[Float](0.0, 5.0)
+	# var p4 = new Point[Float](5.0, 5.0)
+	# var arr = new Array[Point[Float]].with_items(p1, p2, p3, p4)
+	# var poly = new ConvexPolygon.with_vertices(arr)
+	# poly.anti_clock_sort
+	# p1 = new Point[Float](2.5, 2.5)
+	# p2 = new Point[Float](7.5, 2.5)
+	# p3 = new Point[Float](2.5, 7.5)
+	# p4 = new Point[Float](7.5, 7.5)
+	# arr = new Array[Point[Float]].with_items(p1, p2, p3, p4)
+	# var poly2 = new ConvexPolygon.with_vertices(arr)
+	# poly2.anti_clock_sort
+	# assert poly.intersects(poly2)
+	# ~~~
+	fun intersects(other: ConvexPolygon): Bool do
+		assert is_convex
+
+		var axes1 = get_axes
+		var axes2 = other.get_axes
+		for i in [0..axes1.length[ do
+			var axis = axes1[i]
+			var project1 = project(axis)
+			var project2 = other.project(axis)
+			if not project1.overlap(project2) then return false
+		end
+		for i in [0..axes2.length[ do
+			var axis = axes2[i]
+			var project1 = project(axis)
+			var project2 = other.project(axis)
+			if not project1.overlap(project2) then return false
+		end
+		return true
+	end
+
+	# Projection of the axis of the polygon for the intersection test
+	private fun project(axis: Point[Float]): Projection do
+		var min = axis.x * points[0].x + axis.y * points[0].y
+		var max = min
+		for i in [0..points.length[ do
+			var p = axis.x * points[i].x + axis.y * points[i].y
+			if p < min then min = p
+			if p > max then max = p
+		end
+		var projection = new Projection(min, max)
+		return projection
+	end
+
+	# Check if this polygon is convex
```

Comment:
When returning a boolean we try to use a question, it is often much clearer: `Is this polygon convex?`